application-analytics-ui/src/main/resources/Analytics/Code/Macros/RowEvolutionJSON.xml

```diff
@@ -41,13 +41,39 @@
   'period' : $request.period,
   'date' : $request.date,
   'module' : 'API',
-  'method' : 'Actions.getPageUrl',
   'format' : 'json'
 })
+#set ($filters = {})
 #if ($request.macroName == 'MostViewedPages')
```

Comment:
What do you say about refactoring to something like this:
```
#set ($macrosParams = {
  'MostViewedPages': {
    'method': 'Actions.getPageUrl',
    'rowIdentifierName': 'pageUrl',
    'rowIdentifierValue': $request.rowIdentifier
  }, 'Browser': {
    'method': 'DevicesDetection.getModel',
    'rowIdentifierName': 'segment',
    'rowIdentifierValue': $request.rowIdentifier
  },
  ...
})
#set ($parameters = {})
#set ($macroParams = $macrosParams.get($request.macroName))
#set ($discard = $parameters.put($macroParams.rowIdentifierName, $macroParams.rowIdentifierValue))
#set ($discard = $parameters.put('method', $macroParams.method))
```

Reply:
Actually, I noticed now on [PageFollowingASiteSearch](https://github.com/xwikisas/application-analytics/pull/30/files#diff-a03c3eeff3a8ec55bd955ee5dbe8313b607bae39af96eca19f155d8b59aaa385R50) that there can be multiple parameters to add, so this is not actually helpfull